src/adapters/env.ts

```diff
@@ -37,6 +39,7 @@ const envToDomain = ({
 }: Env): Promise<domain.Env> => {
   const polygonZkEVMNetworkId = z.number().positive().parse(Number(VITE_POLYGON_ZK_EVM_NETWORK_ID));
   const useFiatExchangeRates = z.boolean().parse(VITE_USE_FIAT_EXCHANGE_RATES === "true");
+  const isNetworkOutdated = z.boolean().parse(VITE_IS_NETWORK_OUTDATED === "true");
```

Comment:
This is not safe because it results in `false` for any non-valid value, meaning that whenever the env holds a value different from `true` no error will be shown and the network outdated msg won't be shown.

Instead, we should only succeed with `true` or `false` and provide a meaningful error in any other case.

```diff
diff --git a/src/adapters/env.ts b/src/adapters/env.ts
index 952af38..4327bb6 100644
--- a/src/adapters/env.ts
+++ b/src/adapters/env.ts
@@ -38,8 +38,8 @@ const envToDomain = ({
   VITE_USE_FIAT_EXCHANGE_RATES,
 }: Env): Promise<domain.Env> => {
   const polygonZkEVMNetworkId = z.number().positive().parse(Number(VITE_POLYGON_ZK_EVM_NETWORK_ID));
-  const useFiatExchangeRates = z.boolean().parse(VITE_USE_FIAT_EXCHANGE_RATES === "true");
-  const isNetworkOutdated = z.boolean().parse(VITE_IS_NETWORK_OUTDATED === "true");
+  const useFiatExchangeRates = stringBoolean.parse(VITE_USE_FIAT_EXCHANGE_RATES);
+  const isNetworkOutdated = stringBoolean.parse(VITE_IS_NETWORK_OUTDATED);

   return getChains({
     ethereum: {
@@ -122,6 +122,27 @@ const envParser = StrictSchema<Env, domain.Env>()(
     .transform(envToDomain)
 );

+const stringBoolean = StrictSchema<string, boolean>()(
+  z.string().transform((value, context) => {
+    switch (value) {
+      case "true": {
+        return true;
+      }
+      case "false": {
+        return false;
+      }
+      default: {
+        context.addIssue({
+          code: z.ZodIssueCode.custom,
+          fatal: true,
+          message: "The provided string input can't be parsed as a boolean",
+        });
+        return z.NEVER;
+      }
+    }
+  })
+);
+
 const loadEnv = (): Promise<domain.Env> => {
   const parsedEnv = envParser.parseAsync(import.meta.env);
```

Reply:
done!